src/app/reporting/Engine.cpp

```diff
@@ -53,10 +60,11 @@ EventNumber Engine::CountEvents(ReadHandler * apReadHandler, EventNumber * apIni
 }
 
 CHIP_ERROR
-Engine::RetrieveClusterData(AttributeDataElement::Builder & aAttributeDataElementBuilder, ClusterInfo & aClusterInfo)
+Engine::RetrieveClusterData(AttributeDataList::Builder & aAttributeDataList, ClusterInfo & aClusterInfo)
 {
     CHIP_ERROR err                              = CHIP_NO_ERROR;
-    AttributePath::Builder attributePathBuilder = aAttributeDataElementBuilder.CreateAttributePathBuilder();
+    AttributeDataElement::Builder attributeDataElementBuilder = aAttributeDataList.CreateAttributeDataElementBuilder();
```

Comment:
throughout his codebase: `goto exit` forces us to place these declarations on top. In this case we declare them 20 lines before they are actually used :( this makes me very unhappy as a reviewer, needing to always look up and down for scope.

Nothing we can consistently do, however venting some frustration and asking to try hard to stop with the goto exit approach.